frontend/src/editor/types.ts

```diff
@@ -8,12 +8,13 @@ export type Text = {
 export type Paragraph = {
   type: 'paragraph';
   children: Text[];
-  speaker: string;
+  speakers: number[];
```

Comment:
why / how can a paragraph have multiple speakers?

Reply:
Short (unsurprising) answer: There might be more than one person speaking in a paragraph.

Short definition of terminology: A Segment is the output of diarization. It has a start, end and a list of speaker ids.

A Paragraph is the output of transcription. It has a start, end, list of speakers and text children.

The reason to change this here is: The segments and paragraphs do not line up perfectly. There might be multiple segments overlapping one paragraph. The question is how we deal with that. One option would be to split the paragraph to align with the segmetns – but then the same problem occurs with single words (a segment might end/start during a word).

The way the code currently deals with that is: A paragraph gets assigned the speakers of all segments that overlap it. I think this needs more finetuning in the future (for example splitting paragraphs in some cases), but is a good baseline for now.

Reply:
As discussed: I changed the types to have an optional `speaker` (assigned if there is only one possible speaker) and a list of `alternative_speakers`